R/common-find-variation.R

```diff
@@ -425,4 +425,498 @@ getPipedValue <- function(string) {
   } else {
     result
   }
+}
+
+
+#' @title
+#' Find variation across groups
+#' @description Plot a boxplot to compare the variation across the groups. 
+#' @param df A dataframe
+#' @param categoricalCols Vector of strings representing categorical column(s)
+#' @param measureColumn Vector of strings representing measure column(s)
+#' @param printPlot Optional, default is TRUE. FALSE: only shows the statistics  
+#' used to plot the boxplot. 
+#' @param printTable Optional, default is TRUE. FALSE: not to show the table of  
+#' mean/std and quartiles 
+#' @param dateCol Optional. A date(time) column to group by (done by month) 
+#' @return A boxplot to compare variation across groups.
+#' @export
+#' @references \url{http://healthcare.ai}
+#' @seealso \code{\link{healthcareai}} \code{\link{calculateCOV}} 
+#' \code{\link{createVarianceTallTable}}
+#' @examples
+#' df <- data.frame(Gender = factor(rbinom(100, 1, 0.45), labels = c("Male","Female")), 
+#'                  Age = factor(rbinom(100, 1, 0.45), labels = c("Young","Old")),
+#'                  Ans = factor(rbinom(100, 1, 0.45), labels = c("Y","N")),
+#'                  Dept = sample(c("A","B","C"), 50, replace = TRUE, prob = c(0.2,0.3,0.5)),
+#'                  LOS1 = c(rnorm(30),rnorm(70,10,5)),
+#'                  LOS2 = c(rnorm(20,8,3),rnorm(80,-15,5)))
+#'
+#' categoricalCols <- c("Dept","Gender")
+#' measureColumn <- c("LOS1","LOS2")
+#' 
+#' variationAcrossGroups(df = df, 
+#'                        categoricalCols = categoricalCols,
+#'                        measureColumn = measureColumn)
+#'                        
+#' treatment = c(rep("A", 20) , rep("B", 20) , rep("C", 20), rep("D", 20) ,  rep("E", 20))
+#' value = c( sample(2:5, 20 , replace = TRUE) , sample(6:10, 20 , replace = TRUE), 
+#'         sample(1:7, 20 , replace = TRUE), sample(3:10, 20 , replace = TRUE) , 
+#'         sample(10:20, 20 , replace = TRUE) )
+#' df1 = data.frame(treatment,value)
+#' 
+#' variationAcrossGroups(df1, categoricalCols = "treatment", measureColumn = "value")
+#'
+
+
+variationAcrossGroups <- function(df, 
+                                  categoricalCols,
+                                  measureColumn,
+                                  printPlot = TRUE,
+                                  printTable = TRUE,
+                                  dateCol = NULL) {
+  
+  if (!all(c(categoricalCols,measureColumn,dateCol) %in% names(df))) {
+    stop('The measure column or one of the categorical cols is not in the df')
+  }
+  
+  # Check that measure columns exist and are of proper type
```

Comment:
Not a huge deal, but you could combine the two error checks for measureColumn and CategoricalCols by leaving out the ifs and just using `for (i in 1:length(measureColumn))` and `for(i in 1:length(categoricalCols))`